projects/plugins/jetpack/modules/theme-tools/site-logo.php

```diff
@@ -44,3 +39,40 @@ function site_logo_init() {
 	}
 }
 add_action( 'init', 'site_logo_init' );
+
+/**
+ * When switching from a legacy theme that uses `site-logo` to a theme that uses `custom-logo`,
+ * update the theme's custom logo if it doesn't already have one.
+ *
+ * @return void
+ */
+function jetpack_update_custom_logo_from_site_logo() {
+	$site_logo = get_option( 'site_logo' );
+
+	if ( current_theme_supports( 'custom-logo' ) && ! get_theme_mod( 'custom_logo' ) && $site_logo ) {
+		set_theme_mod( 'custom_logo', $site_logo );
```

Comment:
How well does this play with the syncing code in Gutenberg? I think it has to deregister some handlers to make sure that it doesn't get stuck in a loop, so it might be worth checking that this doesn't trigger an update of the `site_logo` option.

Reply:
It does trigger the Gutenberg/WP 5.8 syncing code, but it doesn't have any practical effect:

- Jetpack updates custom_logo theme_mod, from site_logo option
- Gutenberg/Core sync custom_logo theme mod back to site_logo option (which are already the same, anyway)